internal/clients/azure.go

```diff
@@ -107,14 +100,11 @@ func spAuth(ctx context.Context, pc *v1alpha1.ProviderConfig, ps *terraform.Setu
 	if err := json.Unmarshal(data, &azureCreds); err != nil {
 		return errors.Wrap(err, errUnmarshalCredentials)
 	}
+	// set credentials configuration
 	ps.Configuration[keySubscriptionID] = azureCreds[keyAzureSubscriptionID]
-	// set credentials environment
-	ps.Env = []string{
-		fmt.Sprintf(fmtEnvVar, envSubscriptionID, azureCreds[keyAzureSubscriptionID]),
-		fmt.Sprintf(fmtEnvVar, envTenantID, azureCreds[keyAzureTenantID]),
-		fmt.Sprintf(fmtEnvVar, envClientID, azureCreds[keyAzureClientID]),
-		fmt.Sprintf(fmtEnvVar, envClientSecret, azureCreds[keyAzureClientSecret]),
-	}
+	ps.Configuration[keyTenantID] = azureCreds[keyAzureTenantID]
```

Comment:
Could you confirm that these are not printed in the error logs?

Reply:
Hi @muvaf,
I gave this a try by injecting an invalid key in Terraform provider configuration block. Here's what appears in managed resource status:
```yaml
  - message: 'observe failed: cannot run refresh: refresh failed: Extraneous JSON
      object property: No argument or block type is named "non-existent".: File name:
      main.tf.json'
    reason: ReconcileError
    status: "False"
    type: Synced
```

And no relevant provider logs unless debug logging is enabled. If debug logs are enabled, then I think we are dumping Terraform output, which may, in theory leak sensitive data. We can discuss this general issue in Terrajet.